packages/plugins/src/foundry/utils/index.ts

```diff
@@ -382,80 +386,48 @@ export const makeGetConfigArtifacts = (
     const toReadFiles: string[] = []
     const localBuildInfoCache = {}
 
-    const fullyQualifiedNamePromises = fullyQualifiedNames.map(
-      async (fullyQualifiedName) => {
-        const artifact = await readContractArtifact(
-          fullyQualifiedName,
-          projectRoot,
-          artifactFolder
-        )
-
+    const fullyQualifiedNamePromises = initCodeWithArgsArray.map(
+      async (initCodeWithArgs) => {
         // Look through the cache for the first build info file that contains the contract
         for (const file of sortedCachedFiles) {
-          if (file.fullyQualifiedNames?.includes(fullyQualifiedName)) {
+          const contract = file.contracts.find((ct) => {
+            const { bytecode, constructorFragment, linkReferences } = ct
+
+            return isInitCodeMatch(initCodeWithArgs, {
+              bytecode,
+              linkReferences,
+              constructorFragment,
+            })
+          })
+
+          if (contract) {
             // Keep track of if we need to read the file or not
             if (!toReadFiles.includes(file.name)) {
               toReadFiles.push(file.name)
             }
 
+            const artifact = await readContractArtifact(
+              contract.fullyQualifiedName,
+              projectRoot,
+              artifactFolder
+            )
+
             return {
-              fullyQualifiedName,
+              fullyQualifiedName: contract.fullyQualifiedName,
               artifact,
               buildInfoName: file.name,
             }
           }
         }
-
-        // Throw an error if no build info file is found in the cache for this contract
-        // This should only happen if the user manually deletes a build info file
-        if (existsSync(buildInfoCacheFilePath)) {
-          unlinkSync(buildInfoCacheFilePath)
-        }
-        throw new Error(
-          `Build info cache is outdated, please run 'forge build --force' then try again.`
-        )
```

Comment:
It's necessary to remove this in order to support contracts that are defined as inline bytecode, e.g. a `CREATE3` proxy. If we don't remove these lines, [this test case for `getConfigArtifacts`](https://github.com/sphinx-labs/sphinx/pull/1341/files#diff-36de83659b4192824ef2b7a81238450fd670bbe6aadc70b51e979cd332c7c007R601-R609) will throw an error. If the user deletes a build info file, we'll display a warning in the preview saying that we couldn't identify these contracts, which is reasonable I think. If you think it'll be common for users to run into this issue, we could add something to the troubleshooting guide or the warning in the preview

Reply:
I think this is acceptable.